### ISSUE TITLE
S3900 is missing the first usage where something would be broken by a null

### ISSUE THREAD

**Issue by StingyJack** [SonarSource/sonar-dotnet#4613]
Given the code below from a Startup.cs. SonarLint v4.35.0.32570 is not warning about missing the first usage of `env`, but is instead warning about a later usage. I'm going to suppress these with an attribute because `env` really should never be null.

``` csharp
 public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IServiceProvider serviceProvider,
            ILoggerFactory loggerFactory, IHttpContextAccessor httpContextAccessor, LoggerConfig loggerConfig)
        {
            try
            {
                RegisterLogger(loggerFactory, httpContextAccessor, loggerConfig);

                if (env.IsDevelopment())  //sonar is not complaining about this usage
                {
                    app.UseDeveloperExceptionPage();
                }
                else
                {
                    app.UseHsts();
                }

                //serve the protos directory for other potential clients to consume
                var provider = new FileExtensionContentTypeProvider();
                provider.Mappings.Clear();
                provider.Mappings[".proto"] = "text/plain";
                app.UseStaticFiles(new StaticFileOptions
                {
                    FileProvider = new PhysicalFileProvider(Path.Combine(env.ContentRootPath, "Protos")), //Sonar lint is complaining about this usage
                    RequestPath = "/protos",
                    ContentTypeProvider = provider
                });
                app.UseDirectoryBrowser(new DirectoryBrowserOptions
                {
                    FileProvider = new PhysicalFileProvider(Path.Combine(env.ContentRootPath, "Protos")),
                    RequestPath = "/protos"
                });
...
```

**Comment by pavel-mikula-sonarsource** [comment]
Hi @StingyJack, 

`.IsDevelopment()` is an extension method. Those are ignored by S3900 because they can survive `null` value.
